src/lib.rs

```diff
@@ -729,6 +771,86 @@ impl DerInner<'_> {
     }
 }
 
+/// Data encoded according to TLS presentation language, either owned or borrowed.
+///
+/// This wrapper type is used to represent TLS-encoded data in a way that is agnostic to whether
+/// the data is owned (by a `Vec<u8>`) or borrowed (by a `&[u8]`). Support for the owned
+/// variant is only available when the `alloc` feature is enabled.
+///
+/// For more information on "TLS encoding" or the TLS presentation language, see
+/// [RFC 8446 §3](https://www.rfc-editor.org/rfc/rfc8446#section-3).
+#[derive(Clone)]
+pub struct TlsBytes<'a>(TlsBytesInner<'a>);
```

Comment:
How do we feel about renaming `Der` instead to something more generic, so that we don't have to duplicate all this stuff? We can retain `Der` as a type alias for compatibility.

(The next question is bound to be, what do we call it -- `ByteSlice`, `Cow`, `Bytes`? I don't like `Bytes` because it is already a widely used type; `Cow` is too, and also a bit of a misnomer because we're largely treating it as immutable so the copy-on-write behavior isn't so relevant.)

Reply:
Unless I've overlooked something, we could:

- rename the existing `DerInner` type to make it more generic (to some extent the name doesn't matter, because it doesn't leak out the crate -- `BytesInner`? just `Inner`?)
- directly use that type inside `EchConfigBytes`, therefore eliminating the whole `TlsBytes` type? ( A later `Sct` type could take the same approach, and wouldn't require a new intermediate type like `TlsBytes`.)

It seems like `EchConfigBytes` already must have the totality of its interface, because its interior `TlsBytes` is non-pub. I suppose the question is whether the `TlsBytes` type is expected to be useful separately?

(The same argument could be made for the `Der` type, but that's more academic since it's in released versions.)

Reply:
These ideas sound reasonable. My main concern was avoiding a semver break, but I think the proposals above would avoid that.

I don't think the `TlsBytes` type is likely to be super useful on its own. I was mainly pattern matching the existing approach, which as you mention, is perhaps of questionable utility w.r.t `Der` in the public interface.

I'll adjust w/ the proposal Ctz put forward,  `DerInner` -> `BytesInner` and `EchConfigBytes(TlsBytes)` -> `EchConfigBytes(BytesInner)`.

Reply:
Yup, sounds good to me.

Reply:
Done. :ballot_box_with_check: